src/gstring.rs

```diff
@@ -180,6 +180,11 @@ impl Deref for GString {
 impl From<GString> for String {
     #[inline]
     fn from(s: GString) -> Self {
+        if let GString::ForeignOwned(ref cstring) = s {
+            if let Ok(s) = String::from_utf8(cstring.as_bytes().to_vec()) {
```

Comment:
Hm, I missed that we have buildin method
https://doc.rust-lang.org/std/ffi/struct.CString.html#method.into_string

Reply:
Sorry, I don't follow.

Reply:
```
error[E0507]: cannot move out of borrowed content                                                                                                                                                                  
   --> src/gstring.rs:184:28                                                                                                                                                                                       
    |                                                                                                                                                                                                              
184 |             if let Ok(s) = cstring.into_string() {                                                                                                                                                           
    |                            ^^^^^^^ cannot move out of borrowed content                                                                                                                                       
                                                                                                                                                                                                                   
error: aborting due to previous error 
```

Reply:
We want remove copy in foreign owned case.
As [`slice::to_vec`](https://doc.rust-lang.org/std/primitive.slice.html#method.to_vec) make copy,
we need get inner slice and give it to new String with transfer ownership.

Reply:
Yeah, remove the `ref` keyword here

Reply:
That's not possible because GString implements the Drop trait :/

Reply:
Then let's do the standard workaround for that :) Store the `CString` in an `Option<_>` and use `Option::take()` here, and in all other places use `as_ref().unwrap()` when accessing the `Option<_>`